frontend/src/pages/storageClasses/StorageClassDefaultRadio.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Flex, Radio, Spinner, Tooltip } from '@patternfly/react-core';
+import { updateStorageClassConfig } from '~/services/StorageClassService';
+
+interface StorageClassDefaultRadioProps {
+  storageClassName: string;
+  isChecked: boolean;
+  isDisabled: boolean;
+  onChange: () => void;
+}
+
+export const StorageClassDefaultRadio: React.FC<StorageClassDefaultRadioProps> = ({
+  storageClassName,
+  isChecked: isInitialChecked,
+  isDisabled,
+  onChange,
+}) => {
+  const [isChecked, setIsChecked] = React.useState(isInitialChecked);
+  const [isUpdating, setIsUpdating] = React.useState(false);
+  const id = `${storageClassName}-default-radio`;
+
+  // Reset checked state when updating isChecked from parent component
+  React.useEffect(() => {
+    setIsChecked(isInitialChecked);
+  }, [isInitialChecked]);
+
+  const update = React.useCallback(async () => {
+    setIsUpdating(true);
+
+    try {
+      await updateStorageClassConfig(storageClassName, { isDefault: true });
+      await onChange();
+      setIsUpdating(false);
+    } catch {
+      setIsUpdating(false);
+    }
+
+    setIsChecked(true);
+  }, [storageClassName, onChange]);
+
+  const radioInput = React.useMemo(
+    () => (
+      <Radio
+        id={id}
+        name={id}
+        data-testid="set-default-radio"
+        label="Set as default"
+        isChecked={isChecked}
+        isDisabled={isDisabled || isUpdating}
+        onChange={update}
+      />
+    ),
+    [id, isChecked, isDisabled, isUpdating, update],
+  );
+
+  return (
+    <Flex spaceItems={{ default: 'spaceItemsMd' }} alignItems={{ default: 'alignItemsCenter' }}>
+      {isDisabled ? (
+        <Tooltip content="Enable this class to set it as the default.">{radioInput}</Tooltip>
+      ) : (
+        radioInput
+      )}
+
+      <Spinner
+        size="md"
+        aria-label="Loading default radio selection"
+        style={{ visibility: isUpdating ? 'visible' : 'hidden' }}
+      />
```

Comment:
to follow what is done in other areas: disable the checkbox when loading, not need for a spinner

Reply:
This was specifically asked for here by @xianli123

https://www.figma.com/proto/372he4ynUVkovm3lwfFVbg/Storage-classes?page-id=978%3A58598&node-id=1037-64843&node-type=FRAME&viewport=808%2C369%2C0.09&t=ZZSwrwX1rXaSAepA-1&scaling=scale-down-width&content-scaling=fixed

Reply:
@Gkrumbach07 @jpuzz0 How about disabling the radio button when checking it? It will be changed to the normal style after the API request is done and the class table is refreshed. WDYT?

Reply:
@xianli123 this is being done already with the current implementation.

Reply:
Awesome! Thanks! @jpuzz0

Reply:
must of missed that. all good then